File: src/php/milo/tracer/TraceContext.php
<?php
/**
 * Milo_TraceContext
 *
 * @package Milo
 * @author Chris Goffinet <cg@chrisgoffinet.com>
 * @filesource
 */

require_once "../util/Util.php";

/**
 * Milo_TraceContext
 *
 * A trace context holds the trace ids and all spans associated with a trace.
 *
 * @package Milo
 * @author Chris Goffinet <cg@chrisgoffinet.com>
 */
class Milo_TraceContext
{
    /**
     * Trace id
     *
     * @var integer
     */
    private $trace_id = null;

    /**
     * Trace enabled
     *
     * @var boolean
     */
    private $enabled = false;

    /**
     * Trace rpc method
     *
     * @var string
     */
    private $rpc_method = null;

    /**
     * Span id
     *
     * @var long
     */
    private $span_id = 0;

    /**
     * Span parent id
     *
     * @var long
     */
    private $span_parent_id = 0;

    /**
     * List of Milo Spans
     *
     * @var array
     */
    private $spans = array();

    /**
     * Return the trace id
     *
     * @return integer
     */
    public function getTraceID()
    {
        return $this->trace_id;
    }

    /**
     * Set the trace id
     *
     * @return void
     */
    public function setTraceID($id)
    {
      $this->trace_id = $id;
    }

    /**
     * Get the next span id
     *
     * @return long
     */
    public function nextSpanID()
    {
      $this->span_id = Milo_Util::generateID();
      return $this->span_id;
    }

    /**
     * Get the current span id
     *
     * @return long
     */
    public function getSpanID()
    {
      return $this->span_id;
    }

    /**
     * Get the current span parent id
     *
     * @return long
     */
    public function getSpanParentID()
    {
      return $this->span_parent_id;
    }

    /**
     * Add a span to the list
     *
     * @return T_Base_Span
     */
    public function addSpan($span)
    {
        $this->spans[] = $span;
        return $span;
    }

    /**
     * Get the list of spans
     *
     * @return array
     */
    public function getSpans()
    {
        return $this->spans;
    }

}

?>

File: src/py/milo/tracer/trace_context.py
from milo.util.span import generateID

class TraceContext(object):
    def __init__(self):
        self.spans = []
        self.trace_id = None
        self.enabled = False
        self.span_id = 0
        self.span_parent_id = 0
        self.rpc_method = None

    def getRPCMethod(self):
        return self.rpc_method

    def setRPCMethod(self, method):
        self.rpc_method = method

    def nextSpanID(self):
        self.span_id = generateID()
        return self.span_id

    def setSpanID(self, id_):
        self.span_id = id_

    def setSpanParentID(self, id_):
        self.span_parent_id = id_

    def getSpanID(self):
        return self.span_id

    def getSpanParentID(self):
        return self.span_parent_id

    def setEnabled(self, enabled):
        self.enabled = enabled

    def isEnabled(self):
        return self.enabled

    def getTraceID(self):
        return self.trace_id

    def setTraceID(self, trace_id):
        self.trace_id = trace_id

    def addSpan(self, span):
        self.spans.append(span)
        return span

    def getSpans(self):
        return self.spans

    def clear(self):
        self.spans = []
        self.events = []

    def __repr__(self):
        return str([span for span in self.spans])

